### DISCUSSION TITLE
Ifx/ groove .bin files

### DISCUSSION THREAD

**Discussion by ShoveD01** [bangcorrupt/hacktribe#113] [Q&A] [answered]
How are these .bin files installed/added

**Comment by bangcorrupt** [comment, marked as answer]
All of the grooves and FX that currently exist are already in hacktribe.  Importing, exporting and editing them currently requires python programming.  The `main()` function of [e2_groove_template.py](https://github.com/bangcorrupt/e2-scripts/blob/2e03bc7fdac69c2fce1a07a159b7a74dda51989d/e2_groove_template.py) demonstrates how to get, edit and set a groove template.  Storing these to flash will be added later.   [e2sysex.py](https://github.com/bangcorrupt/e2-scripts/blob/2e03bc7fdac69c2fce1a07a159b7a74dda51989d/e2sysex.py) includes functions to `get_ifx` and `set_ifx`; ~~there is not yet an editor for FX~~.  In the future, the [GUI](https://github.com/bangcorrupt/hacktribe/discussions/97) will allow editing grooves and FX.

UPDATE:

See [FX Preset Editing](https://github.com/bangcorrupt/hacktribe-editor/wiki/FX-Preset-Editing).
